pennylane/decomposition/decomposition_graph.py

```diff
@@ -418,7 +441,13 @@ def decomposition(self, op: Operator) -> DecompositionRule:
 class _DecompositionSearchVisitor(DijkstraVisitor):
     """The visitor used in the Dijkstra search for the optimal decomposition."""
 
-    def __init__(self, graph: rx.PyDiGraph, original_op_indices: set[int], lazy: bool = True):
+    def __init__(
+        self,
+        graph: rx.PyDiGraph,
+        original_op_indices: set[int],
+        lazy: bool = True,
+        gate_set: dict = None,
```

Comment:
```suggestion
        graph: rx.PyDiGraph,
        gate_set: dict = None,
        original_op_indices: set[int],
        lazy: bool = True,
```
This is a definitely nitpicking but I'd prefer that this `gate_set` argument (being more fundamental to the algorithm) appears earlier in the list of arguments.

Reply:
Sounds good.